tests/phpunit/tests/Checker/Checks/Plugin_Readme_Check_Tests.php

```diff
@@ -31,6 +31,54 @@ public function test_run_with_errors_no_readme() {
 		$this->assertEquals( 'no_plugin_readme', $warnings['readme.txt'][0][0][0]['code'] );
 	}
 
+	public function test_run_with_errors_invalid_name() {
+		$readme_check  = new Plugin_Readme_Check();
+		$check_context = new Check_Context( UNIT_TESTS_PLUGIN_DIR . 'test-plugin-plugin-readme-errors-invalid-name/load.php' );
+		$check_result  = new Check_Result( $check_context );
+
+		$readme_check->run( $check_result );
+
+		$errors   = $check_result->get_errors();
+		$warnings = $check_result->get_warnings();
+
+		$this->assertNotEmpty( $errors );
+		$this->assertArrayHasKey( 'readme.txt', $errors );
+		$this->assertEquals( 1, $check_result->get_error_count() );
+
+		$this->assertEmpty( $warnings );
+		$this->assertEquals( 0, $check_result->get_warning_count() );
+
+		// Check for invalid name error.
+		$this->assertArrayHasKey( 0, $errors['readme.txt'] );
+		$this->assertArrayHasKey( 0, $errors['readme.txt'][0] );
+		$this->assertArrayHasKey( 'code', $errors['readme.txt'][0][0][0] );
+		$this->assertEquals( 'invalid_plugin_name', $errors['readme.txt'][0][0][0]['code'] );
+	}
+
+	public function test_run_with_errors_empty_name() {
+		$readme_check  = new Plugin_Readme_Check();
+		$check_context = new Check_Context( UNIT_TESTS_PLUGIN_DIR . 'test-plugin-plugin-readme-errors-empty-name/load.php' );
```

Comment:
If we only remove the plugin name from the readme file, the parser does not return an error. 🤔 Maybe it's a bug in the parser itself? I tried this to remove the `no_license` error from these tests.

File: tests/phpunit/testdata/plugins/test-plugin-plugin-readme-errors-empty-name/readme.txt
```
Contributors:      plugin-check
Requires at least: 6.0
Tested up to:      6.1
Requires PHP:      5.6
Stable tag:        n.e.x.t
License:           GPLv2 or later
License URI:       https://www.gnu.org/licenses/gpl-2.0.html
Tags:              testing, security

A simple readme file for testing.
```

Reply:
Yes. That's the bug in the Parser. In your above example Parser takes `Contributors:      plugin-check` as `name` and continues to parse other as headers.

Reply:
In that case could you please open ticket on meta for tracking?

Reply:
Meta ticket for improving parsing of plugin name: https://meta.trac.wordpress.org/ticket/7428